Forecasting module and bases

razzervision opened the pull request:
forecasting is added to model.py and repository.py also made forecasting component and a deployments/docker-compose.yml

razzervision committed:
other changes

razzervision committed:
merge should be done now and backend works.

razzervision committed:
removed leftover comment from models.py

razzervision committed:
added deployments folder and moved docker-compose to it

razzervision committed:
renamed dir data/ to weights/

razzervision committed:
removed rest_api bases

Fenrir-M committed:
feat: Initial version of prediction module

Fenrir-M committed:
feat: set up endpoints for predictions

added classes in preparation for saving data in database
added Containerfile to run forecasting on its own
added endpoints for prediction

Fenrir-M committed:
fix: Added classes to __init__.py and added basic error handling

Fenrir-M committed:
i have made a lot of changes such as updating forecast/data_service.py and added a new bases rest_api

Fenrir-M committed:
the rest_api backend works!!!!!

Fenrir-M committed:
changed name from time-series/forecast/ to time-series/forecasting/ and fix type errors and forecasting/data_service

razzervision committed:
fixed model.py, repository, project/forecasting/pyproject.py

razzervision committed:
added bases forecasting_api

razzervision committed:
add forecasting endpoints not tested.

razzervision committed:
fixed docker-compose but not really and modified forecasting/Containerfile

razzervision committed:
update to docker-compose.yml

razzervision committed:
the backend works now again...

the alembic/versions did not work so we make new versions and moved old
to versions_backup docker-compose.yml works and there is communication
between backend and database.
Changes to be committed:
	new file:   alembic/versions/096d037beb21_now_head.py
	renamed:    alembic/versions/3571e1ef04d0_added_status_column_to_analyses_table.py -> alembic/versions_backup/3571e1ef04d0_added_status_column_to_analyses_table.py
	renamed:    alembic/versions/9115ce03c384_changed_the_name_of_the_analysis_model_.py -> alembic/versions_backup/9115ce03c384_changed_the_name_of_the_analysis_model_.py
	renamed:    alembic/versions/a2a07af82d7b_initial.py -> alembic/versions_backup/a2a07af82d7b_initial.py
	renamed:    alembic/versions/ab4fd44eb440_forcasting_merged_version.py -> alembic/versions_backup/ab4fd44eb440_forcasting_merged_version.py
	new file:   alembic/versions_backup/af4f3f40edc7_.py
	renamed:    alembic/versions/f6fbdc23912f_remove_start_date_from_dataset_model.py -> alembic/versions_backup/f6fbdc23912f_remove_start_date_from_dataset_model.py
	modified:   deployments/docker-compose.yml
	modified:   projects/forecasting/pyproject.toml
	modified:   uv.lock

razzervision committed:
fixed versions to import sqlmodel correctly?

Changes to be committed:
	modified:   alembic/versions/096d037beb21_now_head.py

razzervision committed:
moved model location to weights/forecasting

Changes to be committed:
	modified:   components/time_series/forecasting/prediction.py
	renamed:    weights/lstm_energy_model.keras -> weights/forecasting/lstm_energy_model.keras
	renamed:    weights/lstm_energy_model_v2.keras -> weights/forecasting/lstm_energy_model_v2.keras
	renamed:    weights/lstm_energy_weather_model.keras -> weights/forecasting/lstm_energy_weather_model.keras
	renamed:    weights/scalers.pkl -> weights/forecasting/scalers.pkl

razzervision committed:
fix .editorconfig

Changes to be committed:
	modified:   .editorconfig

razzervision commented:
please merge before you change origin/main. thanks :smiley:

soupglasses commented on .pre-commit-config.yaml:
```diff
@@ -1,49 +1,47 @@
-# See https://pre-commit.com for more information
 default_language_version:
```
Don't change this file

soupglasses commented on alembic/versions_backup/ab4fd44eb440_forcasting_merged_version.py:
```diff
@@ -0,0 +1,114 @@
+    """Upgrade schema."""
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "prediction_results",
```
This seems good! :)

soupglasses commented on bases/time_series/forecasting_api/main.py:
```diff
@@ -0,0 +1,36 @@
+
+app.include_router(forecasting.router, prefix="/forecasting", tags=["forecasting"])
+
+add_pagination(app)
```
Do you use pagination?

soupglasses reviewed (changes requested):
Most of the code seems good.

The `alembic/` is in quite a bad state and needs to get fixed before this can get merged.

Also please don't change the `.pre-commit-config.yaml` and `projects/outlier_detection_api/Containerfile` without need.

soupglasses commented:
Surpasses #223

soupglasses commented:
The alembic file needed in `alembic/versions/ce068111380e_add_prediction_results.py`
```python
"""add prediction_results table

Revision ID: ce068111380e
Revises: 3571e1ef04d0
Create Date: 2025-12-15 18:41:00.629950

"""
from typing import Sequence, Union

from alembic import op
import sqlalchemy as sa
import sqlmodel


# revision identifiers, used by Alembic.
revision: str = 'ce068111380e'
down_revision: Union[str, Sequence[str], None] = '3571e1ef04d0'
branch_labels: Union[str, Sequence[str], None] = None
depends_on: Union[str, Sequence[str], None] = None


def upgrade() -> None:
    """Upgrade schema."""
    op.create_table(
        "prediction_results",
        sa.Column("analysis_id", sa.Integer(), nullable=False),
        sa.Column("time", sa.DateTime(), nullable=False),
        sa.Column("value", sa.Float(), nullable=False),
        sa.ForeignKeyConstraint(
            ["analysis_id"],
            ["timeseries.analyses.id"],
        ),
        sa.PrimaryKeyConstraint("analysis_id", "time"),
        schema="timeseries",
    )

def downgrade() -> None:
    """Downgrade schema."""
    op.drop_table("prediction_results", schema="timeseries")
```

razzervision committed:
fixed comments

razzervision committed:
fixed comments

soupglasses commented on alembic/versions/f6fbdc23912f_remove_start_date_from_dataset_model.py:
```diff
@@ -31,7 +31,8 @@ def downgrade() -> None:
     )
-    op.execute("UPDATE datasets SET start_date = '1970-01-01 00:00:00'")
-    op.alter_column("datasets", "start_date", nullable=False)
+    op.execute("UPDATE timeseries.datasets SET start_date = '1970-01-01 00:00:00'")
```
Why `timeseries.datasets`?

soupglasses commented on deployments/docker-compose.yml:
```diff
@@ -0,0 +1,52 @@
+    depends_on:
+      - db
+    ports:
+      - "8002:8000"
```
`"8002:8002"`?

soupglasses commented on projects/forecasting_api/pyproject.toml:
```diff
@@ -0,0 +1,48 @@
+# This section is needed for building
+
+[tool.polylith.bricks]
+"../../components/time_series/greeting" = "time_series/greeting"
```
Greeting does not exist anymore.

soupglasses commented on pyproject.toml:
```diff
@@ -84,11 +90,13 @@ dev-mode-dirs = ["components", "bases", "development", "."]
-"components/time_series/forecast" = "time_series/forecast"
-"components/time_series/settings" = "time_series/settings"
-"components/time_series/uvicorn_runner" = "time_series/uvicorn_runner"
+"components/time_series/forecasting" = "time_series/forecasting"
```
Is there a reason you are removing settings from here?

razzervision committed:
added test for forecasting/prediction.py also fixed rebase error for pyproject.toml

Changes to be committed:
	modified:   .editorconfig
	renamed:    weights/forecasting/lstm_energy_model.keras -> components/time_series/forecasting/assets/lstm_energy_model.keras
	renamed:    weights/forecasting/lstm_energy_model_v2.keras -> components/time_series/forecasting/assets/lstm_energy_model_v2.keras
	renamed:    weights/forecasting/lstm_energy_weather_model.keras -> components/time_series/forecasting/assets/lstm_energy_weather_model.keras
	renamed:    weights/forecasting/scalers.pkl -> components/time_series/forecasting/assets/scalers.pkl
	modified:   components/time_series/forecasting/prediction.py
	modified:   pyproject.toml
	new file:   test/components/time_series/forecasting/__init__.py
	new file:   test/components/time_series/forecasting/test_prediction.py

razzervision committed:
fixed forecasting_api/pyproject.toml

Changes to be committed:
	modified:   projects/forecasting_api/pyproject.toml

razzervision committed:
changed internal container ports and compose internal ports

	modified:   deployments/docker-compose.yml
	modified:   projects/forecasting_api/pyproject.toml

razzervision commented on deployments/docker-compose.yml:
```diff
@@ -0,0 +1,52 @@
+    depends_on:
+      - db
+    ports:
+      - "8001:8000"
```
does not work because unicorn is set up to load the .env from ../api/.env so the webserver will alwayes run on 0.0.0.0:8000 unless ../api/.env has other settings

razzervision committed:
comment out the environment from compose file.

razzervision committed:
remove environment from compose

soupglasses commented on deployments/docker-compose.yml:
```diff
@@ -17,8 +17,8 @@ services:
-    environment:
-      DATABASE_URL: postgresql://${DATABASE_USERNAME}:${DATABASE_PASSWORD}@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_NAME}
+    # environment:
+    #   DATABASE_URL: postgresql://${DATABASE_USERNAME}:${DATABASE_PASSWORD}@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_NAME}
```
Why comment it out? Its not needed anymore. So just remove it.
